R/shiny_fuse.R

```diff
@@ -0,0 +1,718 @@
+      }
+    })
+
+    # Load annoFuse data file --------------------------------------------------
+    observeEvent(input$annofusedatasel, {
+      message("Reading in...")
+      values$annofuse_tbl <- read.delim(input$annofusedatasel$datapath)
+      values$enhanced_annofuse_tbl <- values$annofuse_tbl
+
+      # enhancing the content of the table
+      values$enhanced_annofuse_tbl$Gene1A <- .multilink(values$enhanced_annofuse_tbl$Gene1A)
+      values$enhanced_annofuse_tbl$Gene1B <- .multilink(values$enhanced_annofuse_tbl$Gene1B)
+
+      if (!is.null(values$data_pfam)) {
+        message("Creating domain information...")
+        values$ann_domain <- annoFuse::getPfamDomain(
+          standardFusioncalls = values$annofuse_tbl,
+          bioMartDataPfam = values$data_pfam, # must be pre-loaded
+          # partial overlapping domains are retained == "Partial" with keepPartialAnno=TRUE;
+          # if keepPartialAnno=FALSE then domain retained status == "No"
+          keepPartialAnno = TRUE
+        )
+      }
+    })
+
+    # Main interactive table for exploration -----------------------------------
+    output$table_annofuse <- DT::renderDataTable({
+      validate(
+        need(
+          !is.null(values$annofuse_tbl),
+          "Please upload the results of annoFuse to start the exploration"
+        )
+      )
+
+      DT::datatable(
+        values$enhanced_annofuse_tbl,
+        style = "bootstrap",
+        rownames = FALSE,
+        filter = "top",
+        selection = "single",
+        escape = FALSE,
+        options = list(
+          scrollX = TRUE,
+          pageLength = 25,
+          lengthMenu = c(5, 10, 25, 50, 100, nrow(values$enhanced_annofuse_tbl))
+        )
+      )
+    })
+
+    # TODO? link to the DB where the info was taken from
+
+    # Gene info and plots from TableExplorer
+    output$geneinfo_ui <- renderUI({
+      validate(
+        need(
+          length(input$table_annofuse_rows_selected) > 0,
+          "Please select a row to display the genes info"
+        )
+      )
+
+      row_id <- input$table_annofuse_rows_selected
+      message(row_id)
+      gene_for_content <- values$annofuse_tbl[row_id, "Gene1A"]
+      gene_for_content_2 <- values$annofuse_tbl[row_id, "Gene1B"]
+
+
+      doublegeneinfo_2_html(gene_for_content, gene_for_content_2)
+      # geneinfo_2_html(gene_for_content)
+    })
+
+    output$geneplots_ui <- renderUI({
+      validate(
+        need(
+          length(input$table_annofuse_rows_selected) > 0,
+          message = ""
+        )
+      )
+
+      tagList(
+        h4("Some general info"),
+        tabsetPanel(
+          tabPanel(
+            "Plot left",
+            plotOutput("geneplots_left")
+          ),
+          tabPanel(
+            "Plot right",
+            plotOutput("geneplots_right")
+          )
+        )
+      )
+    })
+
+    # Breakpoint plots ---------------------------------------------------------
+    output$geneplots_right <- renderPlot({
```

Comment:
JL was also wondering if we can add a 'Save' or 'Download' button for the plots?

Reply:
Guess so, yes.
One can at any moment also just right click and save the png, it is in the end nothing but a browser page.
The button might need to handle the plots object slightly differently, but I can check that out!

Reply:
Thanks for looking into this! Yeah I agree with the "right click -> save as" is an easy way to get the plots as well